sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3503,6 +3503,13 @@ pi_result piContextGetInfo(pi_context Context, pi_context_info ParamName,
     return ReturnValue(pi_uint32(Context->Devices.size()));
   case PI_CONTEXT_INFO_REFERENCE_COUNT:
     return ReturnValue(pi_uint32{Context->RefCount.load()});
+  case PI_EXT_ONEAPI_CONTEXT_INFO_USM_MEMCPY2D_SUPPORT:
+    // 2D USM memcpy is supported.
+    return ReturnValue(pi_bool{true});
+  case PI_EXT_ONEAPI_CONTEXT_INFO_USM_FILL2D_SUPPORT:
+  case PI_EXT_ONEAPI_CONTEXT_INFO_USM_MEMSET2D_SUPPORT:
+    // 2D USM fill and memset is not supported.
+    return ReturnValue(pi_bool{false});
```

Comment:
https://github.com/intel/llvm/pull/7370/files#diff-e6666c7a4c98da6b3e0a5393de2d1f176f90aff0d2db2775d8e6b5c8e6d14152R73: 
how a single SYCL feature macro is sufficient to check these 3 distinct capabilities?

Reply:
All three operations are under the same extension ([sycl_ext_oneapi_memcpy2d](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_memcpy2d.asciidoc)) with the same SYCL feature macro. We could potentially split the extension up so we have finer granularity in SYCL implementation support, but with PR we are able support all three operations, either directly in PI or through fallback kernel implementations so I do not think we necessarily need the finer granularity.

Reply:
If the single feature test macro says support is there, but 2 of 3 operations are reported as not supported, then there is a problem I believe. Now I guess I missed this part:
> with PR we are able support all three operations, either directly in PI or through fallback kernel implementations

So, we can still support it, right?

Reply:
Yes, exactly! From the user's perspective all backends support all 3 operations, but behind the scenes the functions query the backends to see which path to take.

Reply:
Cool, thanks!